### PR TITLE
fix(nvhttp): wrap TLS socket to ensure graceful closure

### PR DESCRIPTION
**Pull request by cgutman**
## Description
The upstream `SimpleWeb::Server<SimpleWeb::HTTPS>` implementation doesn't ensure the TLS session is torn down gracefully which leads to unexpected TCP RST packets from Sunshine to the client when using TLS v1.3 and setting `close_connection_after_response` (which NvHTTP does). Depending on the client, this spurious TCP RST can lead to the HTTPS request failing even though all the required data did end up on the wire. The `QSslSocket` implementation in Qt seems to be particularly susceptible to the issue, causing computers to constantly bounce between online and offline.

Ideally this would be fixed upstream, but the way that the classes are structured with `SimpleWeb::ServerBase` doesn't provide a straightforward place to insert this code (`ServerBase` has no clue about any TLS stuff). Fixing it on our side using a little wrapper class is fairly simple with most of the diff just being renames of the old class to the new one. We already had a class derived from `SimpleWeb::Server<SimpleWeb::HTTPS>`, so changing it to derive from `SimpleWeb::ServerBase` just required a few more lines from the upstream constructor of `SimpleWeb::Server<SimpleWeb::HTTPS>`.

## Type of Change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency update (updates to dependencies)
- [ ] Documentation update (changes to documentation)
- [ ] Repository update (changes to repository files, e.g. `.github/...`)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated the in code docstring/documentation-blocks for new or existing methods/components

### CODE REVIEW

File: src/nvhttp.cpp [hunk 1]
```diff
@@ -44,19 +44,38 @@ namespace nvhttp {
...
+      // Gracefully shutdown the TLS connection
+      SimpleWeb::error_code ec;
+      shutdown(ec);
+    }
+  };
+
+  class SunshineHTTPSServer: public SimpleWeb::ServerBase<SunshineHTTPS> {
+  public:
+    SunshineHTTPSServer(const std::string &certification_file, const std::string &private_key_file):
+        ServerBase<SunshineHTTPS>::ServerBase(443),
```
Comment (ns6089) [inline src/nvhttp.cpp:62]: Is it ok to hardcode the port?

Comment (cgutman) [reply src/nvhttp.cpp:62]: Yeah, it's just the default port number for the socket type. It's always been there, but it was just previously hidden inside the constructor for `SimpleWeb::ServerBase<SimpleWeb::HTTPS>` before.

**Comment by Hazer** [comment]
@cgutman Any migration steps for current users running this behind some reverse proxy or it's mostly transparent?

**Comment by cgutman** [comment]
It's transparent to any reverse proxies.

**Pull request merged**